circle/src/point.rs

```diff
@@ -0,0 +1,165 @@
+use core::ops::{Add, AddAssign, Mul, Neg, Sub};
+
+use p3_field::extension::ComplexExtendable;
+use p3_field::{ExtensionField, Field};
+
+/// Affine representation of a point on the circle.
+/// x^2 + y^2 == 1
+// _private is to prevent construction so we can debug assert the invariant
+#[allow(clippy::manual_non_exhaustive)]
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
+pub struct Point<F> {
+    pub x: F,
+    pub y: F,
+    _private: (),
+}
+
+impl<F: Field> Point<F> {
+    #[inline]
+    pub fn new(x: F, y: F) -> Point<F> {
+        debug_assert_eq!(x.square() + y.square(), F::one());
+        Point { x, y, _private: () }
+    }
+    pub fn zero() -> Self {
+        Self::new(F::one(), F::zero())
+    }
+
+    /// Circle STARKs, Section 3, Lemma 1: (page 4 of the first revision PDF)
```

Comment:
We could refer to the first (and only) version on IACR via https://eprint.iacr.org/archive/2024/278/20240219:113204.